specs/ChainSpec/NetworkSpec.md

```diff
@@ -0,0 +1,119 @@
+# Network
+
+Network layer constitutes the lower level of the NEAR protocol and is ultimately responsible of transporting messages between peers. To provide an efficient routing it maintains a routing table between all peers actively connected to the network, and send message between them using best paths. There is a mechanism in place that allows new peers joining the network to discover other peers, and rebalance network connections in such a way that latency is minimized. Cryptographic signatures are used to check identities from peers participating in the protocol since it is non-permissioned system.
+
+This document should serve as reference for all the clients to implement networking layer.
+
+## Messages
+
+All data structure used for messages between peers are enumerated in [Message](Messages.md).
+
+## Discovering the network
+
+When a node start for the first time it tries to connect to a list of bootstrap nodes specified via a config file.
+
+It is expected that a node on a regular basis request a list of peers from its neighboring nodes to learn about other nodes in the network. This will allow every node to discover each other, and have relevant information to try to establish a new connection with it. When a node receive a message of type [`PeersRequest`](Messages.md#PeersMessage) it is expected to answer with a message of type [`PeersResponse`](Message.md#PeersMessage) with information from healthy peers known to this node. Both of this message are unrestricted, i.e. can be sent at any moment without incurring in a fault, but beware about the policy to [ban for abusive behavior](#Abusive-behavior).
+
+### Handshakes
+
+To establish new connections between pair of nodes, they will follow the following protocol. Node A open a connection with node B and sends a [Handshake](Messages.md#Handshake) to it. If handshake is valid (see reasons to [decline the handshake](#Decline-handshake)) then node B will proceed to send [Handshake](Messages.md#Handshake) to node A. After each node accept a handshake it will mark the other node as an active connection, until one of them stop the connection.
+
+[Handshake](Messages.md#Handshake) contains relevant information about the node, the current chain and information to create a new edge between both nodes.
+
+#### Decline handshake
+
+When a node receive a handshake from other node it will decline this connection if one of the following situations happens:
+
+1. Other node has different genesis.
+2. Edge nonce is too low
+
+#### Edge
+
+Edges are used to let other know in the network that there is currently an active connection between a pair of nodes. See the signature of [this data structure](Messages.md#Edge).
```

Comment:
"to let other know in the network" -> "to let others in the network know"
"the signature of this datastructure" -> "the definition of this datastructure" (to avoid confusion with cryptographic signatures)